packages/@aws-cdk/core/lib/stack.ts

```diff
@@ -424,6 +424,17 @@ export class Stack extends CoreConstruct implements ITaggable {
     return CloudFormationLang.toJSON(obj, space).toString();
   }
 
+  /**
+   * DEPRECATED
+   * @deprecated use `reportMissingContextKey()`
+   */
+  public reportMissingContext(report: cxapi.MissingContext) {
+    if (!Object.values(cxschema.ContextProvider).includes(report.provider as cxschema.ContextProvider)) {
+      throw new Error(`Unknown context provider requested in: ${JSON.stringify(report)}`);
+    }
```

Comment:
As-is, these checks will be removed in v2. Is that intentional?

Reply:
I believe this is only needed because `cxapi.MissingContext` needs to be converted into `cxschema.MissingContext`.
Specifically, the `provider` prop is a `string` in the former and an `enum` in the latter.
Hence not required in the new API.

cc @rix0rrr to confirm who introduced [the change](https://github.com/aws/aws-cdk/commit/87860cac6cc8b57eca29cb32c986331e5d62e883#diff-caa054ee92a558f0c56b9e350128d032b8504f4d30193f70a9a8a4a2e707c990).